.github/workflows/linux-clang.yaml

```diff
@@ -0,0 +1,27 @@
+name: Build for Linux with Clang
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'Branch or commit to build'
+        required: true
+        default: 'main'
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Clang
+      run: sudo apt-get install -y clang
```

Comment:
Should this end up in a Docker image rather than installing per run? (Happy to do in a second PR if that's easier).

Reply:
I will remove it as it is already part of the on-demand GitHub runner.